sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2475,18 +2476,21 @@ checkDevSupportDeviceRequirements(const device &Dev,
   }
 
   // Checking if device supports defined required work group size
-  if (ReqdWGSizePropIt) {
-    ByteArray ReqdWGSize =
-        DeviceBinaryProperty(*(ReqdWGSizePropIt.value())).asByteArray();
+  if (ReqdWGSizeUint32TPropIt || ReqdWGSizeSizeTPropIt) {
+    bool usingSizeT = ReqdWGSizeSizeTPropIt.has_value();
+    auto it = usingSizeT ? ReqdWGSizeSizeTPropIt : ReqdWGSizeUint32TPropIt;
+
+    ByteArray ReqdWGSize = DeviceBinaryProperty(*(it.value())).asByteArray();
     // Drop 8 bytes describing the size of the byte array.
     ReqdWGSize.dropBytes(8);
-    uint64_t ReqdWGSizeAllDimsTotal = 1;
-    std::vector<uint64_t> ReqdWGSizeVec;
+    size_t ReqdWGSizeAllDimsTotal = 1;
```

Comment:
This is one more tricky use of a `size_t`. Once again, we are trying to match `size_t` selected by the toolchain for user application.

However, unlike with the compiler, I doubt that we can have runtime compiler with `size_t` size different from user app, because we likely pass arguments of that type between headers and the library, i.e. it is likely a part of our ABI anyway.

Therefore, I don't think that any action is required here, but personally, I would still prefer to see an explicit fixed-width data type.